sorting/insertion_sort_recursive.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file
+ * @brief Insertion Sort Algorithm
+ *
+ * @details
+ * Insertion sort is a simple sorting algorithm that builds the final
+ * sorted array one element at a time. It is much less efficient compared
+ * to other sorting algorithms like heap sort, merge sort, or quick sort.
+ * 
+ * However, it has several advantages:
+ * - Easy to implement.
+ * - Efficient for small data sets.
+ * - More efficient than other O(n²) algorithms like selection sort or bubble sort.
+ * - Stable: it does not change the relative order of elements with equal keys.
+ * 
+ * Insertion sort works similarly to how people sort playing cards in their hands.
+ * The algorithm iterates through the list and inserts each element into its correct 
+ * position in the sorted portion of the array.
+ *
+ * Example execution:
+ * 1. Start with the array [4, 3, 2, 5, 1].
+ * 2. Insert 3 in its correct position: [3, 4, 2, 5, 1].
+ * 3. Insert 2: [2, 3, 4, 5, 1].
```

Comment:
Please add references/sources for further reading

Reply:
Added the requested changes you asked for.
My contribution is for Hacktober fest.